### PR TITLE
feat(e2e): Add `monitor` command to manage Prometheus and Grafana servers

### PR DESCRIPTION
**Pull request by hvanz**
Ref https://github.com/cometbft/cometbft/pull/4087#discussion_r1756569658 

This PR adds a new command `monitor` with subcommands `start` and `stop`.

For example, `./build/runner -f <manifest> monitor start` will start a Docker container with Prometheus and Grafana, reading metrics from the nodes defined in the manifest. Before running this command, run `./build/runner -f <manifest> setup` to generate a `prometheus.yml` config file.

By default, Grafana will start with two predefined dashboards called Main and Compact.

---

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [X] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: test/e2e/monitoring/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+```bash
+docker compose up -d
+```
+
+To stop all monitoring services:
+```bash
+docker compose down
+```
+
+# Details
```
Comment (melekes) [inline test/e2e/monitoring/README.md:24]:
```suggestion
## Details
```

File: test/e2e/runner/setup.go [hunk 2]
```diff
@@ -120,7 +125,13 @@ func Setup(testnet *e2e.Testnet, infp infra.Provider) error {
 	}
 
 	if testnet.Prometheus {
-		if err := testnet.WritePrometheusConfig(); err != nil {
+		if err := WritePrometheusConfig(testnet, PrometheusConfigFile); err != nil {
+			return err
+		}
+		// Make a copy of the Prometheus config file in the testnet direactory.
```
Comment (melekes) [inline test/e2e/runner/setup.go:131]:
```suggestion
		// Make a copy of the Prometheus config file in the testnet directory.
```

**Review by melekes — APPROVED** [review]
🚀

**Review by jmalicevic — APPROVED** [review]

**Review by andynog — APPROVED** [review]
looks good to me, tested locally and it worked as expected.

**Comment by andynog** [comment]
@hvanz if possible, could we back port this to `v1.x` ? Might be helpful in troubleshooting e2e nightly errors

**Comment by hvanz** [comment]
@Mergifyio backport v1.x

**Comment by hvanz** [comment]
>  if possible, could we back port this to v1.x ? Might be helpful in troubleshooting e2e nightly errors

@andynog Done: https://github.com/cometbft/cometbft/pull/4409

**Pull request merged**